cli/cook/subcommands/list.py

```diff
@@ -70,16 +75,51 @@ def show_data(cluster_job_pairs):
     print_info(job_table)
 
 
+def date_time_string_to_ms_since_epoch(date_time_string):
+    """Converts the given date_time_string (e.g. '5 minutes ago') to milliseconds since epoch"""
+    calendar = parsedatetime.Calendar()
+    local_tz = tzlocal.get_localzone()
+    dt, parse_status = calendar.parseDT(datetimeString=date_time_string, tzinfo=local_tz)
+    if parse_status == 1 or parse_status == 2:
+        ms_since_epoch = int((dt - EPOCH).total_seconds() * 1000)
+        logging.debug(f'parsed "{date_time_string}" as {dt} (ms = {ms_since_epoch}) using time zone {local_tz}')
+        return ms_since_epoch
+    else:
+        raise Exception(f'"{date_time_string}" is not a valid date / time string.')
+
+
+def lookback_hours_to_range(lookback_hours):
+    """Converts the given number of hours to a start and end range of milliseconds since epoch"""
+    end_ms = int(round(time.time() * 1000))
+    lookback_ms = int(lookback_hours * MILLIS_PER_HOUR)
+    start_ms = end_ms - lookback_ms
+    logging.debug(f'converted lookback {lookback_hours} to start ms {start_ms} and end ms {end_ms}')
+    return start_ms, end_ms
+
+
 def list_jobs(clusters, args):
     """Prints info for the jobs with the given list criteria"""
     as_json = args.get('json')
-    state = args.get('state')
+    states = args.get('states')
     user = args.get('user')
     lookback_hours = args.get('lookback')
+    submitted_after = args.get('submitted_after')
+    submitted_before = args.get('submitted_before')
     name = args.get('name')
     limit = args.get('limit')
 
-    query_result = query(clusters, state, user, lookback_hours, name, limit)
+    if lookback_hours and (submitted_after or submitted_before):
+        raise Exception('You cannot specify both lookback hours and submitted after / before times.')
```

Comment:
This pattern seems to be repeated in a couple of places. As far as I can tell, these checks are really enforcing mutually exclusive command-line arguments. Maybe we should do that at the level of `argparse` instead? 
https://docs.python.org/3/library/argparse.html#mutual-exclusion

Reply:
The mutual exclusion groups are nice, but they don't seem to support "either this one, or either or both of these other two" (you can either do lookback, or submitted after and/or submitted before).

Reply:
I think you can nest groups, which would let you do that. But let's just ask this question first: Assuming that the restriction *can* be made at the level of `argparse`, does it make more sense to do it in `argparse` or in the sub-command logic?

Reply:
OK, it actually looks like nested groups aren't officially supported right now (see https://stackoverflow.com/a/43288171/1427124), so we're stuck with putting the logic elsewhere.

Reply:
Yeah, I tried nesting a group within the mutual exclusion group, and it didn't work :(